File: js/src/gradingStudents.js
function gradingStudents(grades) {
	// Write your code here
	return grades.map((grade) => {
		if (grade >= 38) {
			const nextGradeMultiple = Math.ceil(grade / 5) * 5;
			return nextGradeMultiple - grade < 3 ? nextGradeMultiple : grade;
		} else {
			return grade;
		}
	});
}

module.exports = gradingStudents;

File: java/src/main/java/org/hackerrank/GradingStudents.java
package org.hackerrank;

import java.util.ArrayList;
import java.util.List;

public class GradingStudents {
    public static List<Integer> gradingStudents(List<Integer> grades) {
        List<Integer> roundedGrades =  new ArrayList<>();
        for (int grade:
             grades) {
            int roundedGrade = (int)Math.ceil(((double) grade / 5)) * 5;
            if (grade >= 38 && roundedGrade - grade < 3) {
                roundedGrades.add(roundedGrade);
            } else {
                roundedGrades.add(grade);
            }
        }
        return roundedGrades;
    }
}
